### PR TITLE
Create simple Sentry example for exception reporting.

### PR DESCRIPTION
**Pull request by leerob**
## Overview

For many teams using Next.js, setting up exception monitoring is a critical requirement before going to production. The current [with-sentry](https://github.com/zeit/next.js/pull/5727) example, while [thorough](https://github.com/zeit/next.js/pull/5727#issuecomment-446007181), is very complicated for beginners. 

This PR adds a simple example showing how to use [Sentry](https://sentry.io) to catch & report errors on both client and server-side.

- `_document.js` is _server-side only_ and is used to change the initial server-side rendered document markup. We listen at the node process level to capture exceptions.
- `_app.js` is client-side only and is used to initialize pages. We use the `componentDidCatch` lifecycle method to catch uncaught exceptions.

**Note**: Source maps will not be sent to Sentry when running locally. It's also possible you will see duplicate errors sent when testing locally due to hot reloading. For a more accurate simulation, please deploy to Now.

## Testing
With the example deployed the Now, I can click the links to simulate errors.
image
Then, I'm able to see the errors flow into Sentry.
image
The errors are properly source mapped.
image

@timneutkens

### CODE REVIEW

**Comment by MarchWorks** [comment]
can you check this while you are at it https://github.com/zeit/next.js/issues/6783

**Comment by leerob** [comment]
@MarchWorks I can confirm that is not an issue with this example. I navigate back successfully here.

**Comment by MarchWorks** [comment]
@leerob run `yarn lint-fix` to fix some of these issues if not all

**Comment by leerob** [comment]
@MarchWorks Good catch! Thanks for that.

**Comment by sheerun** [comment]
Your example won't catch errors from getInitialProps of App

**Comment by leerob** [comment]
> Your example won't catch errors from getInitialProps of App

Yeah, I think that's okay. If that is absolutely necessary, then they should probably use the more complex example.

**Comment by sheerun** [comment]
Maybe add comment that warns about this?

File: examples/with-sentry-simple/pages/index.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+class Index extends React.Component {
+  static getInitialProps ({ query }) {
+    if (query.raiseError) {
+      throw new Error('Error in getInitialProps')
+    }
+  }
+
+  state = {
+    raiseError: false
```
Comment (kris-luminar) [inline examples/with-sentry-simple/pages/index.js]:
```suggestion
        raiseErrorInRender: false,
        raiseErrorInUpdate: false
```

Comment (kris-luminar) [reply examples/with-sentry-simple/pages/index.js]: Pretty sure this is what you wanted here.

**Comment by pruhstal** [comment]
@leerob curious if this could take advantage of `_error.js` per @timneutkens comment: https://github.com/zeit/next.js/pull/5727#discussion_r235981700 from previously.

**Comment by leerob** [comment]
@pruhstal I view overriding `error.js` as a nice to have. We should include the client side Sentry code in `_app.js` instead of just `_error.js` since `App` wraps every client-side page. 

Having a custom error page would be a nice addition, but I wanted to keep this example as simple as possible.

**Comment by pruhstal** [comment]
@leerob fair enough. Are you using this setup currently in a production build anywhere? Am also curious how difficult it makes it not having a session id attached to each error (per the current `with-sentry`) example.

**Comment by leerob** [comment]
@pruhstal We are. So far, haven't seen any major issues. The first application consuming this setup isn't too complicated (Mostly static Next.js frontend + custom Hapi server). If we do run into issues with this setup, I'll plan on updating this example.

**Comment by lfades** [comment]
@leerob I'm okay with this example, can you resolve the suggestion made by @kris-luminar first ? 🙏

**Comment by leerob** [comment]
@lfades Addressed. I'm not seeing merge conflicts locally, so I'm a little confused where those are coming frome.

**Comment by leerob** [comment]
Ahh, I merged `master` instead of `canary`. Addressed.

**Review by leerob — COMMENTED** [review]
It seems like Prettier formatting was just now picked up for some of the examples 🤷‍♂

**Comment by lfades** [comment]
@leerob The prettier changes are cool and can be included in the future but in a PR made for it, please remove the unnecessary prettier changes and focus on the example 🙏

**Comment by leerob** [comment]
@lfades Closing in favor of https://github.com/zeit/next.js/pull/7360 👍

**Pull request closed without merging**